image/image.go

```diff
@@ -0,0 +1,255 @@
+// setting a number of max retires for the HTTP client
+func NewAPIClientWithMaxRetries(imageAPIURL string, maxRetries int) *Client {
+	hcClient := healthcheck.NewClient(service, imageAPIURL)
+	if maxRetries > 0 {
+		hcClient.Client.SetMaxRetries(maxRetries)
+	}
+
+	return &Client{
+		cli: hcClient.Client,
+		url: imageAPIURL,
+	}
+}
+
+// Checker calls image api health endpoint and returns a check object to the caller.
+func (c *Client) Checker(ctx context.Context, check *health.CheckState) error {
+	hcClient := healthcheck.Client{
+		Client: c.cli,
+		URL:    c.url,
+		Name:   service,
+	}
+
+	return hcClient.Checker(ctx, check)
+}
+
+// GetImages returns the list of images
+func (c *Client) GetImages(ctx context.Context) (m Images, err error) {
+	uri := fmt.Sprintf("%s/images", c.url)
+
+	clientlog.Do(ctx, "retrieving images", service, uri)
+
+	req, err := http.NewRequest(http.MethodGet, uri, nil)
+	if err != nil {
+		return
+	}
+
+	resp, err := c.cli.Do(ctx, req)
+	if err != nil {
+		return
+	}
+	defer closeResponseBody(ctx, resp)
+
+	if resp.StatusCode != http.StatusOK {
+		err = NewImageAPIResponse(resp, uri)
+		return
+	}
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return
+	}
+
+	if err = json.Unmarshal(b, &m); err != nil {
+		return
+	}
+
+	return
+}
+
+// PostImage performs a 'POST /images' with the provided NewImage
+func (c *Client) PostImage(ctx context.Context, data NewImage) (m Image, err error) {
+	payload, err := json.Marshal(data)
+	if err != nil {
+		return
+	}
+
+	uri := fmt.Sprintf("%s/images", c.url)
+
+	clientlog.Do(ctx, "posting new image", service, uri)
+
+	req, err := http.NewRequest(http.MethodPost, uri, bytes.NewReader(payload))
+	if err != nil {
+		return
+	}
+
+	resp, err := c.cli.Do(ctx, req)
+	if err != nil {
+		return
+	}
+	defer closeResponseBody(ctx, resp)
+
+	if resp.StatusCode != http.StatusOK {
+		err = NewImageAPIResponse(resp, uri)
+		return
+	}
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return
+	}
+
+	if err = json.Unmarshal(b, &m); err != nil {
+		return
+	}
+
+	return
+}
+
+// GetImage returns a requested image
+func (c *Client) GetImage(ctx context.Context, imageID string) (m Image, err error) {
+	uri := fmt.Sprintf("%s/images/%s", c.url, imageID)
+
+	clientlog.Do(ctx, "retrieving images", service, uri)
+
+	req, err := http.NewRequest(http.MethodGet, uri, nil)
+	if err != nil {
+		return
+	}
+
+	resp, err := c.cli.Do(ctx, req)
+	if err != nil {
+		return
+	}
+	defer closeResponseBody(ctx, resp)
+
+	if resp.StatusCode != http.StatusOK {
+		err = NewImageAPIResponse(resp, uri)
+		return
+	}
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return
+	}
+
+	if err = json.Unmarshal(b, &m); err != nil {
+		return
+	}
+
+	return
+}
+
+// PutImage updates the specified image
+func (c *Client) PutImage(ctx context.Context, imageID string, data Image) (m Image, err error) {
+	payload, err := json.Marshal(data)
```

Comment:
If we decide to use the iota enum, here we could verify that the provided Image state is a valid state